src/main/java/com/commercetools/sync/commons/utils/QueryAll.java

```diff
@@ -164,7 +164,7 @@ private void mapOrConsume(@Nonnull final List<T> pageElements) {
     @Nonnull
     private CompletionStage<PagedQueryResult<T>> queryPage(@Nullable final QueryPredicate<T> queryPredicate) {
         final QueryDsl<T, C> query = baseQuery.withLimit(pageSize);
```

Comment:
As discussed maybe we should always take queryPredicate and not assume that it might be null. It also means that here https://github.com/commercetools/commercetools-sync-java/pull/464/files#diff-eebb1f732a21746c9d39441c25721442L177 you also do not pass a null but users query.

Reply:
thanks for the idea: 52918d7

Reply:
approved f2f.